### DISCUSSION TITLE
graphics library

### DISCUSSION THREAD

**Discussion by npxp** [deepmind/open_spiel#1186] [General]
Hi, I was watching the introductory video you made on YouTube. At one point in the video, the existence of a graphics library for viewing games was mentioned, contributed by Daniel hennes, if I'm not mistaken. How can I access it? Is this available on openspiel?

**Comment by lanctot** [comment]
Hi, 

I'm not 100% sure what you mean by graphics library, but if I mentioned Daniel Hennes, then I was likely referring to the nice phase portrait of evolutionary dynamics or to the automated games visualizations, which are:

- here: https://github.com/google-deepmind/open_spiel/tree/master/open_spiel/python/egt (visualization.py), and
- here: https://github.com/google-deepmind/open_spiel/tree/master/open_spiel/python/visualizations.

If you are looking for visualizations of the games themselves, there's a nice external project called [SpielViz](https://github.com/michalsustr/spielviz) created by @michalsustr.

**Reply by npxp** [reply]
![image](https://github.com/google-deepmind/open_spiel/assets/163313496/76c966a7-44b3-416d-a0f7-28dd8a1a287a)
I'm referring to this. Thanks for the links, I will study more about them

**Reply by lanctot** [reply]
Ok the file that generated it was the second link in my response above.

**Reply by light188** [reply]
Hi, I encountered some difficulties while trying to run SpielViz after installing it. When I tried to run it, I kept receiving an error message: AttributeError: module 'pyspiel' has no attribute 'GameParameter' 
Is this caused by a version update of OpenSpiel? If that's the case, which version should I use to adapt to SpielViz?

**Reply by GeorgeBreahna** [reply]
Had the same problem. [Opened an issue for visibility.](https://github.com/google-deepmind/open_spiel/issues/1224)

**Reply by lanctot** [reply]
Hi, great thanks, but this is a bug in SpielViz not open_spiel. It needs to be fixed on their end. Can you please open the same bug on the SpielViz page instead? Thanks.

**Comment by kvr06-ai** [comment]
If you're looking for a way to visualize OpenSpiel game trees, [spielviz](https://github.com/kvr06-ai/spielviz) covers this — it's a lightweight matplotlib-based library.

```
pip install spielviz
```

```python
import spielviz
spielviz.quick_render("kuhn_poker", path="kuhn.png")
```

It renders game trees with player-colored nodes, action labels, information set highlighting, and strategy profile overlays. This is currently an MVP — if it's useful, I'm planning to add normal-form game rendering, compact tree layouts, learning dynamics plots, and interactive HTML export. Feedback welcome: https://github.com/kvr06-ai/spielviz/issues

Repo: https://github.com/kvr06-ai/spielviz

**Comment by kvr06-ai** [comment]
Interactive demo notebook: [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/kvr06-ai/spielviz/blob/main/examples/spielviz_demo.ipynb)
